hypershift-operator/controllers/hostedcluster/network_policies.go

```diff
@@ -201,24 +201,36 @@ func reconcileKASNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyp
 		}
 
 		if len(managementClusterNetwork.Spec.ClusterNetwork) > 0 {
-			clusterNetworks := []string{}
-			for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
-				clusterNetworks = append(clusterNetworks, network.CIDR)
-			}
+			var allowedClusterNetworkCIDR string
 
 			// Allow to any destination not on the management cluster service network
 			// i.e. block all inter-namespace egress from KAS not allowed by other rules
-			policy.Spec.Egress = append(policy.Spec.Egress,
-				networkingv1.NetworkPolicyEgressRule{
-					To: []networkingv1.NetworkPolicyPeer{
-						{
-							IPBlock: &networkingv1.IPBlock{
-								CIDR:   "0.0.0.0/0",
-								Except: clusterNetworks,
+			for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
+				ipv4, err := hyperutil.IsIPv4(network.CIDR)
```

Comment:
I think we should group all ipv4 CIDRs and all ipv6 CIDRs and create a single entry for each, otherwise if you have multiple ipv4 CIDRs I don't think the rules will have the desired effect.

Reply:
You're right, I've added the new code already, @csrwng PTAL if you have time